### PR TITLE
feat(model): fix type casting and enable bfloat16 optimum-rbln compiled models

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

This PR
- fixes precision issue when model's orignal precision and compiled model's precision are different
- enables bfloat16 models in vllm-rbln
---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`bug-fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: vllm_rbln/model_executor/models/optimum/model_base.py [hunk 1]
```diff
@@ -172,6 +172,16 @@ def init_model(self) -> None:
         self.attn_impl = model.get_attn_impl() if hasattr(
             model, "get_attn_impl") else None
 
+    @property
+    def dtype(self) -> torch.dtype:
+        assert self.model is not None
+        _torch_dtype = self.model.rbln_config._torch_dtype
```
[the remark this answers is not in the document]
Comment (rebel-jonghewk) [reply vllm_rbln/model_executor/models/optimum/model_base.py]: @rebel-eunji @rebel-seinpark would `rbln_config` always have `_torch_dtype` including previous versions?

[the remark this answers is not in the document]
Comment (rebel-seinpark) [reply vllm_rbln/model_executor/models/optimum/model_base.py]: @rebel-jonghewk yes. `_torch_dtype` will be maintained for backward compatibility.

**Review by rebel-jaehwang — APPROVED** [review]

File: vllm_rbln/model_executor/models/optimum/whisper.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def forward(self, model_input: ModelInputForRBLN,
             # OR just generate pooled tensor in the model instance?
             decoder_attention_mask = torch.zeros(self.batch_size,
                                                  self.dec_max_seq_len,
-                                                 dtype=torch.float32)
+                                                 dtype=self.dtype)
```
Comment (rebel-eunji) [inline vllm_rbln/model_executor/models/optimum/whisper.py]: @rebel-kblee @rebel-seinpark 
nit:
This isn’t an issue since whisper currently supports only torch.float32.
However, if whisper becomes available in other data types, please let me know so I can update the logic accordingly.
The `decoder_attention mask` is typically boolean, but for now I’m assuming that it uses torch.float32 for computation.
